### DISCUSSION TITLE
How to deal with custom model in 1.1.0-RC version

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1061] [Q&A]
Hi,
   I used glsp 1.0.0 before,  my custom model has custom properties just like the tasklist demo, there is CreateNodeOperationHandler in that version, and I can extend CreateNodeOperationHandler to handle my custom model.  
   now I want to migrate to the 1.1.0-RC version, but it seems that CreateNodeOperationHandler has been removed in the new version, only GModelCreateNodeOperationHandler is left. 
   should I create a CreateNodeOperationHandler by myself ?  or we should extend all custom model handles from GModelCreateNodeOperationHandler? I am not sure about the center idea of the 1.1.0-RC version of GModel,

**Comment by srcnuzn** [comment]
@wss29 This related thread might be helpful to you : https://github.com/eclipse-glsp/glsp/discussions/1067

**Reply by wss29** [reply]
hi,
   thanks for your reply. I am not sure whether GModel will be used instead of my custom model, #1067 did not give the answer

**Reply by rsoika** [reply]
What did you mean with GModel? In RC10 you have to use the new class `org.eclipse.glsp.server.gmodel.GModelCreateOperationHandler` as the base class for your Create-Node-Operations. The GModel architecture has not changed. So it is only a minor change needed in your code to provide the new method `createCommand` instead of the old one `executeOperation`.  But maybe I get you totally wrong...

**Reply by wss29** [reply]
Hi,
     1. My custom model does not extend from GNode. I use server-node, not Java server, my model format is like follow
              export interface ModelA {
                  prop1: string;
                  prop2: string;
                  prop3: string;
              }
version 1.0.0 has a CreateNodeOperationHandler, all my create handles extend it. follow is an example
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/634d8c98-e24f-4a5f-85df-cf1297aeb10b)
I extend a custom GModelFactory to transform my model to GNode.
version 1.1.0 RC10 remove CreateNodeOperationHandler, instead use GModelCreateNodeOperationHandler as the base handle. GModelCreateNodeOperationHandler needs you to implement createNode in the extended handle. Is that mean I do not need my 
custom GModelFactory?

what I have done before is like the GLSPExample tasklist demo.  what you have done is like GLSPExample workflow demo

after seeing the 1.1.0-RC10 code. I am thinking the tasklist and workflow now become the same situation. 
my custom model is SModel, and I must transform it to GModel first and then do other things. all thing is like workflow demo.

if all is based on GModel in version 1.1.0, first I need to create a new GNode type for each kind of my custom model.  and then implement createNode for each custom createhandle. because I have many custom models, changing them needs time.

I am not very sure of the direction.  so hope the glsp developing team give some suggestion.

**Reply by rsoika** [reply]
@wss29  sorry with the node-server impl I can't help you. I am not a expert in this.
